cognitedata/reveal
# feat(react-components): Add tool and domain obect for making Image360Annotation

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:

## Description :pencil:

* Added tool, domain object and creator for making points inside a 360 image.
* This reuse `LineDomainObject` 
* Fix error in Hover. New using a `pointerEvent`
* On `LineDomainObject `, make it possible to override a transformer that transform the points.
* This is because Image360Annotation stores the points as vectors from center of the image. So to get correct coordinates everywhere `getTransformedPoint `and  `getCopyOfTransformedPoint `was made. The default implementation works as before.

Also:
* Remove `Points and Polyline` classes. Not in use
* Fixed some minor errors in length calculations for closed polygon in `LineDomainObject`
* Added some new property to the calling code easer: pointCount, lineSegmentCount, isClosed, firstPoint and lastPoint. Use them in order to make the code easier to read and reuse.

## How has this been tested? :mag:

The new features need to be tested by go inside a 360 image and use the polygon tool in storybook.

- Goto storybook/architecture
- Go into a 360 image
- The 4th icon in the toolbar (polygon) will appear.
- Make it active
- Add some points
- You can add several polygons
- Try to select one and delete it
- Go out, when the visualization will be hidden.
- Go in again. the visualization will be there.

## Test instructions :information_source:




## Checklist :ballot_box_with_check:
